lib/carbon/hashing.py

```diff
@@ -89,7 +89,8 @@ def remove_node(self, key):
     self.ring_len = len(self.ring)
 
   def get_node(self, key):
-    assert self.ring
+    if not self.ring:
+      raise AssertionError("self.ring not set")
```

Comment:
I'd move this check to the end of the constructor. No point checking this on every get_node(). The whole idea of assertions is that they are optimised away in production code but you can use it while developing/debugging.

Reply:
I didn't realize that we were deliberately trying to optimise away assertions.

Reply:
This code is 10 years old, no clue what was the intention. I also have no clue why Deepsource thinks having assertions in code outside of tests is a bad thing. I personally don't use them but there are many reasons one might want to, there is many CS research on that topic.

Anyway this looks good to me.

Reply:
Ah, but now tests fail... Aaargh

Reply:
We can probably just remove the whole check, it is kinda pointless, if ring = None then the bisect_left() fails anyway.